src/common.speech/SpeakerIdMessageAdapter.ts

```diff
@@ -0,0 +1,162 @@
+import {
+    IRequestOptions,
+    IRestResponse,
+    RestConfigBase,
+    RestMessageAdapter,
+    RestRequestType,
+} from "../common.browser/Exports";
+import {
+    createNoDashGuid,
+    Deferred,
+    IAudioSource,
+    Promise,
+    PromiseResult,
+} from "../common/Exports";
+import {
+    PropertyId,
+    SpeakerIdentificationModel,
+    SpeakerVerificationModel,
+    VoiceProfile,
+    VoiceProfileType,
+} from "../sdk/Exports";
+import { SpeakerRecognitionConfig } from "./Exports";
+
+/**
+ * Implements methods for speaker recognition classes, sending requests to endpoint
+ * and parsing response into expected format
+ * @class SpeakerIdMessageAdapter
+ */
+export class SpeakerIdMessageAdapter {
+    private privRestAdapter: RestMessageAdapter;
+    private privUri: string;
+
+    public constructor(config: SpeakerRecognitionConfig) {
+
+        const connectionId: string = createNoDashGuid();
+        let endpoint = config.parameters.getProperty(PropertyId.SpeechServiceConnection_Endpoint, undefined);
+        if (!endpoint) {
+            const region: string = config.parameters.getProperty(PropertyId.SpeechServiceConnection_Region, "westus");
+            const host: string = config.parameters.getProperty(PropertyId.SpeechServiceConnection_Host, "https://" + region + ".api.cognitive.microsoft.com/speaker/{mode}/v2.0/{dependency}");
```

Comment:
although speaker recognition is not available in azure.cn for now, it's better to support azure.cn regions here. You can refer [here](https://github.com/microsoft/cognitive-services-speech-sdk-js/blob/master/src/common.speech/SpeechConnectionFactory.ts#L46)

Reply:
Updated to support azure.cn if chinese region detected